atum/src/test/scala/za/co/absa/atum/AtumImplicitsSpec.scala

```diff
@@ -63,4 +63,71 @@ class AtumImplicitsSpec extends AnyFlatSpec with SparkTestBase with Matchers {
     spark.disableControlMeasuresTracking()
   }
 
+  "method getControlMeasure" should "return ControlMeasure object" in {
+    fs.delete(new Path(outputPath), false)
+
+    // Initializing library to hook up to Apache Spark
+    spark.enableControlMeasuresTracking(Some(inputPath), None)
+      .setControlMeasuresWorkflow("getControlMeasure")
+
+    import spark.implicits._
+    val df = spark.read.json(Seq(inputData).toDS)
+
+    // act
+    val controlMeasure = df.getControlMeasure
+
+    // assert
+    assert(controlMeasure == inputControlMeasure)
+
+    fs.delete(new Path(outputPath), false)
+    spark.disableControlMeasuresTracking()
+  }
+
+  "method getAllAdditionalInfo" should "return Map[String, String] with given info" in {
+    fs.delete(new Path(outputPath), false)
+
+    // Initializing library to hook up to Apache Spark
+    spark.enableControlMeasuresTracking(Some(inputPath), None)
+      .setControlMeasuresWorkflow("getControlMeasure")
+
+    import spark.implicits._
+    val df = spark.read.json(Seq(inputData).toDS)
+
+    val additionalInfoNoInfo = df.getAllAdditionalInfo
+    additionalInfoNoInfo should equal (Map())
+
+    df.setAdditionalInfo(("key1", "value1"))
+    df.setAdditionalInfo(("key2", "value2"))
+    val expectedAdditionalInfo = Map("key1" -> "value1", "key2" -> "value2")
+
+    val additionalInfo = df.getAllAdditionalInfo
+    additionalInfo should equal (expectedAdditionalInfo)
+
+    fs.delete(new Path(outputPath), false)
+    spark.disableControlMeasuresTracking()
+  }
+
+  "method getAdditionalInfo(key: String)" should "return value for the given key" in {
+    fs.delete(new Path(outputPath), false)
+
+    // Initializing library to hook up to Apache Spark
+    spark.enableControlMeasuresTracking(Some(inputPath), None)
+      .setControlMeasuresWorkflow("getControlMeasure")
+
+    import spark.implicits._
+    val df = spark.read.json(Seq(inputData).toDS)
+    df.setAdditionalInfo(("key1", "value1"))
+    df.setAdditionalInfo(("key2", "value2"))
+
+    // act
+    val additionalInfoValueExistingKey = df.getAdditionalInfo("key2")
+    val additionalInfoValueNonExistingKey = df.getAdditionalInfo("noSuchKey")
+
+    // assert
+    additionalInfoValueExistingKey should equal (Option("value2"))
```

Comment:
```suggestion
    additionalInfoValueExistingKey should equal (Some("value2"))
```

Again, no point in using the `apply` method here, even though it clearly results in `Some("value2")` as well, because "value2" is not `null`.
```scala
object Option {
  def apply[A](x: A): Option[A] = if (x == null) None else Some(x)
}
```
The real instances of `Option[String]` are only `None` and `Some(stringHere)`, there is no point in not using those directly.

Reply:
It's interesting, I actually lived under the belief, that is always preferable to use `Option(value)` and `Some(value)` having most sense only on matching. I recall we talked about it on one CQC, but as I remember Georgi taking part in the discussion, actually presenting the preference for `Option(value)`, it was already some time ago.
Anyway, the reason beside `null` safety had something to do with type casting. Because `Some(value)` is actually differnt type from `Option(value)` (particularly successor).
We might want to revisit the conversation 😉

Reply:
Ok, seems like more of a matter of preference or philosophical discussion. Moreover, in this case where it works the same.